stream_alert_cli/outputs/handler.py

```diff
@@ -62,12 +65,13 @@ def output_handler(options, config):
     if output_exists(output_config, props, service):
         return output_handler(options, config)
 
-    secrets_bucket = '{}.streamalert.secrets'.format(prefix)
-    secrets_key = output.output_cred_name(props['descriptor'].value)
-
-    # Encrypt the creds and push them to S3
-    # then update the local output configuration with properties
-    if not encrypt_and_push_creds_to_s3(region, secrets_bucket, secrets_key, props, kms_key_alias):
+    provider = OutputCredentialsProvider(config=output.config,
+                                         defaults={},  # defaults are unnecessary for saving
```

Comment:
if defaults are 'optional', you can make this an optional keyword argument in the constructor of `OutputCredentialsProvider`